libs/media-player/src/lib/audio-annotator.tsx

```diff
@@ -0,0 +1,274 @@
+};
+
+const isBarVisible = (stateBar: TimeRangeSelectionBar, { timeRangeSelectionBar }: UXState) =>
+    timeRangeSelectionBar === stateBar;
+
+interface AudioAnnotatorProps {
+    audioUrl: string;
+    mimeType?: AudioMIMEType;
+    onTimeRangeSelected: (timeRangeSelected: TimeRangeSelection | null) => void;
+}
+
+const StyledAudioPlayer = styled('audio')`
+    border-radius: 20px;
+`;
+
+const TimeRangeSelectionVisual = styled(Box)({
+    height: '20px',
+    width: '150px',
+    backgroundColor: '#75ecff',
+    borderRadius: '4px',
+    position: 'absolute',
+});
+
+export const AudioAnnotator = ({
+    audioUrl,
+    mimeType,
+    onTimeRangeSelected,
+}: AudioAnnotatorProps) => {
+    const [uxState, setUxState] = useState<UXState>({
+        timeRangeSelectionBar: null,
+    });
+
+    const audioRef = useRef<HTMLVideoElement>(null);
+
+    const [isPlayHeadMoved, setIsPlayHeadMoved] = useState<boolean>(false);
+
+    const [inPointSeconds, setInPointSeconds] = useState<number | null>(null);
+
+    const [outPointSeconds, setOutPointSeconds] = useState<number | null>(null);
+
+    const [errorMessage, setErrorMessage] = useState<string>('');
+
+    const onTimeUpdate = () => {
+        const currentTime = audioRef.current!.currentTime;
+
+        if (currentTime > 0) {
+            setIsPlayHeadMoved(true);
+        }
+    };
+
+    const markInPoint = () => {
+        setErrorMessage('');
+
+        console.log({ inpoint: audioRef.current!.currentTime });
+
+        setInPointSeconds(audioRef.current!.currentTime);
+
+        toggleTimeRangeSelectionBars('inPointSelected');
+    };
+
+    const markOutPoint = () => {
+        setErrorMessage('');
+
+        console.log({ outpoint: audioRef.current!.currentTime });
+
+        const timeRangeSelection = {
+            inPointSeconds: inPointSeconds as number,
+            outPointSeconds: audioRef.current!.currentTime as number,
+        };
+
+        const duration = audioRef.current!.duration;
+
+        if (!isValidTimeRangeSelection(timeRangeSelection, duration)) {
+            if (timeRangeSelection.inPointSeconds >= duration) {
+                setInPointSeconds(null);
+
+                setOutPointSeconds(null);
+
+                setUxState({ timeRangeSelectionBar: null });
+
+                onTimeRangeSelected(null);
+
+                setErrorMessage(
+                    'Not a valid time range.  Marker points cannot be set after play ends'
+                );
+            } else {
+                setErrorMessage('Not a valid time range.  Out point must come after in point');
+            }
+
+            return;
+        }
+
+        setOutPointSeconds(audioRef.current!.currentTime);
+
+        toggleTimeRangeSelectionBars('fullSelection');
+
+        onTimeRangeSelected(timeRangeSelection);
+    };
+
+    const clearMarkers = () => {
+        setInPointSeconds(null);
+
+        setOutPointSeconds(null);
+
+        setUxState({ timeRangeSelectionBar: null });
```

Comment:
See the denormalization here? A lot of your effort is in maintaining consistency of denormalized state.